File: frontEnd/src/types/domain/api/requestModel/SqlExtraction.ts
/**
 * SQL 数据提取配置
 * 从 SQL 查询结果中提取指定行指定列的值，存入变量池供后续步骤使用。
 *
 * 示例：提取 SQL 结果第 1 行 id 列的值
 *   { variableName: "userId", columnName: "id", rowIndex: 0 }
 */
export class SqlExtraction {
    /** 变量名，提取成功后存入变量池的 key */
    variableName?: string;
    /** SQL 结果列名 */
    columnName?: string;
    /** JSON 路径（选填）：列值为 JSON 时按 JSONPath 继续向下提取，如 $.data.id；为空取整列值 */
    jsonPath?: string;
    /** 行下标，0 = 第一行 */
    rowIndex?: number;
    /** 默认值，提取失败时使用 */
    defaultValue?: string;
    /** 是否禁用 */
    disabled?: boolean;
    /** 描述 */
    description?: string;
}

File: backEnd/platform-automation/src/main/java/com/mokatest/platform/demos/api/domain/requestModel/SqlExtraction.java
package com.mokatest.platform.demos.api.domain.requestModel;

import lombok.Data;

/**
 * SQL 数据提取配置
 * 从 SQL 查询结果中提取指定行指定列的值，存入变量池供后续步骤使用。
 *
 * @author JingLong
 * @since 2026-07-28
 */
@Data
public class SqlExtraction {
    /** 变量名 */
    private String variableName;
    /** SQL 结果列名 */
    private String columnName;
    /**
     * JSON 路径（选填）：当列值为 JSON 字符串/JSON 列时，按 JSONPath 继续向下提取嵌套值。
     * 语法与 HTTP 接口提取一致，如 $.data.id。为空则取整列值。
     */
    private String jsonPath;
    /** 行下标，0 = 第一行 */
    private Integer rowIndex;
    /** 默认值 */
    private String defaultValue;
    /** 是否禁用 */
    private boolean disabled = false;
    /** 描述 */
    private String description;
}
